cylc/uiserver/authorise.py

```diff
@@ -0,0 +1,477 @@
+                    Authorization.LIMIT,
+                    default)
+            if isinstance(permission, str):
+                limits.add(permission)
+            else:
+                limits.update(permission)
+        limits.discard('')
+        return limits
+
+    def get_access_user_permissions_from_owner_conf(
+            self, access_user: Dict[str, Union[str, Sequence[Any]]]) -> set:
+        """
+        Returns set of operations specific to access user from owner user conf.
+
+        Args:
+            access_user: Dictionary containing info about access user and their
+            membership of system groups. Defaults to None.
+        """
+        items_to_check = ['*', access_user['access_username']]
+        items_to_check.extend(access_user['access_user_groups'])
+        allowed_operations = set()
+        for item in items_to_check:
+            permission = self.owner_auth_conf.get(item, '')
+            if isinstance(permission, str):
+                allowed_operations.add(permission)
+            else:
+                allowed_operations.update(permission)
+        allowed_operations.discard('')
+        return allowed_operations
+
+    def _get_permitted_operations(self, access_user: str):
+        """Return permitted operations for given access_user.
+
+        Cached for efficiency.
+        Checks:
+        - site config to ensure owner is permitted to give away permissions
+        - user config for authorised operations related to access_user and
+          their groups
+        - if user not in user config, then returns defaults from site config.
+
+        Args:
+            access_user: username to check for permitted operations
+
+        Returns:
+            Set of operations permitted by given access user for this UI Server
+        """
+        access_user_dict = {'access_username': access_user,
+                            'access_user_groups': get_groups(access_user)
+                            }
+        limits_owner_can_give = self.get_owner_site_limits_for_access_user(
+            access_user=access_user_dict)
+        user_conf_permitted_ops = (
+            self.get_access_user_permissions_from_owner_conf(
+                access_user=access_user_dict)
+        )
+        # If not explicit permissions for access user in owner conf then revert
+        # to site defaults
+        if len(user_conf_permitted_ops) == 0:
+            user_conf_permitted_ops = (
+                self.return_site_auth_defaults_for_access_user(
+                    access_user=access_user_dict))
+        user_conf_permitted_ops = expand_and_process_access_groups(
+            user_conf_permitted_ops)
+        limits_owner_can_give = expand_and_process_access_groups(
+            limits_owner_can_give)
+        allowed_operations = limits_owner_can_give.intersection(
+            user_conf_permitted_ops)
+        print(f"inside this method _get_permitted_ops for {access_user}....")
```

Comment:
Should we use logging here? Or was this used for debugging/development?

Reply:
Thanks, will remove. I was testing the caching of the method and forgot to remove.